apps/arweave/src/ar_http_iface_middleware.erl

```diff
@@ -1697,7 +1697,7 @@ search_in_block_index(H, BI) ->
 %% @doc Find a block, given a type and a specifier.
 find_block(<<"height">>, RawHeight) ->
 	BI = ar_node:get_block_index(whereis(http_entrypoint_node)),
-	ar_storage:read_block(binary_to_integer(RawHeight), BI);
+	ar_storage:read_block(RawHeight, BI);
```

Comment:
Hey @goomario, thank you for looking into it.

This way if you ask for /height/2, it will search for a block with the hash `<<"2">>`, we need to do smth like [this](https://github.com/ArweaveTeam/arweave/blob/master/apps/arweave/src/ar_http_iface_middleware.erl#L1668) here instead.

Reply:
Maybe use this:
```
find_block(<<"height">>, RawHeight) when is_integer(RawHeight) ->
	BI = ar_node:get_block_index(whereis(http_entrypoint_node)),
	ar_storage:read_block(RawHeight, BI);
find_block(<<"height">>, RawHeight) ->
	BI = ar_node:get_block_index(whereis(http_entrypoint_node)),
	ar_storage:read_block(binary_to_integer(RawHeight), BI);
```

Reply:
It's work fine, I tested.

Reply:
`find_block(<<"height">>, RawHeight) when is_integer(RawHeight) ->` - this should never match as RawHeight is a binary.